sporkfish/searcher/searcher.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+import chess
+
+# should be absolute paths - amend in next PR
+from ..board.board import Board
+from ..evaluator import Evaluator
+from ..statistics import Statistics
+from .searcher_config import SearcherConfig, SearchMode
+
+# _manager = Manager()
+# We explicitly do not lock these and let race conditions happen
+# Locks are too slow, need to consider atomic maps / values later
+# This might cause underestimation of statistics.
+# _dict = _manager.dict()
+# _stats = _manager.Value("i", 0)
+
+_dict: dict = dict()
+_stats = 0
+
+
+class Searcher(ABC):
+    """
+    Dynamic best move searching class.
+    """
+
+    def __init__(
+        self, evaluator: Evaluator, config: SearcherConfig = SearcherConfig()
+    ) -> None:
+        """
+        Initialize the Searcher instance with mutable statistics.
+
+        :param evaluator: The chess board evaluator.
+        :type evaluator: evaluator.Evaluator
+        :param max_depth: The maximum search depth for the minimax algorithm.
+                         Def.ault is 5.
+        :type max_depth: int
+        :param config: Config to use for searching.
+        :type mode: SearcherConfig
+        :return: None
+        """
+
+        self._evaluator = evaluator
+        self._config = config
+
+        self._statistics = Statistics(_stats)
+
+    def _logging(
```

Comment:
Discussed offline, perhaps a better name would be "log_info" or "log_statistics" ("info" would be in line with UCI info string).

Reply:
Agree with naming - amended